packages/vscode-extension/src/project/project.ts

```diff
@@ -59,13 +59,59 @@ import { EnvironmentDependencyManager } from "../dependency/EnvironmentDependenc
 import { Telemetry } from "./telemetry";
 import { EditorBindings } from "./EditorBindings";
 import { saveMultimedia } from "../utilities/saveMultimedia";
-import { hasAccessToProFeatures, LicenseState, LicenseStatus } from "../common/License";
+import {
+  Feature,
+  FeatureAvailabilityStatus,
+  getFeatureAvailabilityStatus,
+  getLicensesForFeature,
+  LicenseState,
+  LicenseStatus,
+} from "../common/License";
 import { RestrictedFunctionalityError } from "../common/Errors";
 
 const DEEP_LINKS_HISTORY_KEY = "deep_links_history";
 
 const DEEP_LINKS_HISTORY_LIMIT = 50;
 
+/**
+ * Checks if the current license status allows access to the specified feature.
+ * @argument feature - The feature to check access for.
+ * @throws {RestrictedFunctionalityError} if the user does not have access to Pro features
+ */
+function guardFeatureAccess(
+  feature: Feature,
+  shouldSkipLicenseCheck?: (...args: any[]) => boolean
+) {
+  return function (target: any, propertyKey: string, descriptor: PropertyDescriptor) {
+    const originalMethod = descriptor.value;
+    descriptor.value = function (this: Project, ...args: any[]) {
+      if (shouldSkipLicenseCheck) {
+        const shouldSkip = shouldSkipLicenseCheck(args);
```

Comment:
You probably wanted to do
```suggestion
        const shouldSkip = shouldSkipLicenseCheck(...args);
```
Then the callbacks can accept the same arguments as the wrapped function, rather than the array.